packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -141,41 +164,23 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
 
   useGridApiEventHandler(apiRef, GridEvents.resize, handleResize);
 
-  const getRenderableIndexes = ({
-    firstIndex,
-    lastIndex,
-    buffer,
-    minFirstIndex,
-    maxLastIndex,
-  }: {
-    firstIndex: number;
-    lastIndex: number;
-    buffer: number;
-    minFirstIndex: number;
-    maxLastIndex: number;
-  }) => {
-    return [
-      clamp(firstIndex - buffer, minFirstIndex, maxLastIndex),
-      clamp(lastIndex + buffer, minFirstIndex, maxLastIndex),
-    ];
-  };
-
   const updateRenderZonePosition = React.useCallback(
     (nextRenderContext: GridRenderContext) => {
-      const [firstRowToRender] = getRenderableIndexes({
+      const [firstRowToRender, lastRowToRender] = getRenderableIndexes({
         firstIndex: nextRenderContext.firstRowIndex,
         lastIndex: nextRenderContext.lastRowIndex,
         minFirstIndex: 0,
-        maxLastIndex: currentPage.range?.lastRowIndex || 0,
+        maxLastIndex: currentPage.rows.length,
         buffer: rootProps.rowBuffer,
       });
 
-      const [firstColumnToRender] = getRenderableIndexes({
-        firstIndex: nextRenderContext.firstColumnIndex,
-        lastIndex: nextRenderContext.lastColumnIndex,
-        minFirstIndex: renderZoneMinColumnIndex,
-        maxLastIndex: renderZoneMaxColumnIndex,
-        buffer: rootProps.columnBuffer,
+      const firstColumnToRender = getFirstColumnIndexToRender({
```

Comment:
In `getRows` you reused the value from `getRenderableIndexes`. Here you created a function, that internally calculates `getRenderableIndexes` and pass it `getFirstNonSpannedColumnToRender`. Can you do the same thing from `getRows` here (= remove this function)?

Reply:
Sure

Reply:
I got rid of `getRenderableIndexes` intentionally, since it calculates both first and last column index to render.
For `getFirstNonSpannedColumnToRender` I only need first index.

The only reason I've used `getFirstNonSpannedColumnToRender` in `getRows` is that both first and last column indices are calculated there anyway.

Reply:
Ok, no problem.
Is it OK to use `getFirstColumnIndexToRender` in `useGridColumnHeaders` though?